src/redux/slice/ticketsSlice.ts

```diff
@@ -0,0 +1,35 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+import { ITicketData } from "../../types/data";
+
+interface TicketsState {
+  data: ITicketData[] | null;
+  loading: boolean;
+  error: string | null;
+}
+
+const initialState: TicketsState = { data: null, loading: false, error: null };
```

Comment:
[Note]: sometimes, better to use same type default state values, it depends on handling (If from UI side, uninitialised and empty response states are different UIs). For example default data value can be just empty array [].

(this is just info for you, in this case its fine)